src/basic/terminal-util.h

```diff
@@ -158,3 +159,32 @@ int vt_default_utf8(void);
 int vt_reset_keyboard(int fd);
 int vt_restore(int fd);
 int vt_release(int fd, bool restore_vt);
+
+static inline void get_log_colors(int priority, const char **on, const char **off, const char **highlight) {
+        /* Note that this will initialize output variables only when there's something to output.
+         * The caller must pre-initalize to "" or NULL as appropriate. */
```

Comment:
why not initialize the return params to "" in the other cases? sounds easy enough and makes this fully robust to misuse, as it always initializes everything

Reply:
Because one caller wants "" and the other NULL. It felt strange to set it to "" and then do strempty() to undo this in the caller. In addition, the caller will want to skip the call to this function if colors are disabled, so the variables will have to be initialized anyway.

I think this is OK. If we ever use this function in more than two places, the design can be changed.